### PR TITLE
Handle API requests containing blank exclusion values

### PR DESCRIPTION
**Pull request by higs4281**
In version 0.6, the app could generate API requests with single or multiple
blank exclusion values, like these:

```
/admin/autocomplete/search/?query=mortgage&type=ask_cfpb.AnswerPage&exclude=
/admin/autocomplete/search/?query=mortgage&type=ask_cfpb.AnswerPage&exclude=,,
```

The CFPB's page models use integers for primary keys, and these calls produced uncaught
server errors when a database query attempted to check whether query
candidates had keys contained in a blank string. This also disabled the admin forms.

The blank requests were caused by a mix of 'pk' and 'id' variables in js code,
and by a faulty try/except clause in wagtailautocomplete/views.py that allowed
the blanks to pass to a database query.

## Testing

When I tested this patch inside the CFPB's [website project](https://github.com/cfpb/cfgov-refresh), our autocomplete forms
worked again and server errors went away.

I added python tests for wagtail-autocomplete's search view, and the altered
js files pass the npm/jest tests. There are a bunch of Draftail jest tests that fail,
but I think they are outside scope of this PR.

The full tox matrix also completes. ~~, but there are errors in some of the Python3.8
tests. I think those also are outside scope here. Happy to revisit any of that if I'm
misinterpreting test results.~~ **The Python3.8 issue was local to my Mac. The full matrix passes.**

**Notes**: This addresses issue #78 
- I had to update node-sass to get the project to run.
- I also updated the wagtail-autocomplete release version in package.json
- To keep the PR small, I didn't address linting beyond the files changed.

### CODE REVIEW

**Comment by chigby** [comment]
Hi, and thanks for submitting this PR. It looks great! I did have one question about the tests for the search view.

File: wagtailautocomplete/views.py [hunk 1]
```diff
@@ -72,11 +72,9 @@ def search(request):
         queryset = queryset.live()
 
     exclude = request.GET.get('exclude', '')
-    try:
-        exclusions = [unquote(item) for item in exclude.split(',')]
+    if exclude:
+        exclusions = [unquote(item) for item in exclude.split(',') if item]
```
Comment (chigby) [inline wagtailautocomplete/views.py:76]: Does this change prevent an exception in the case of blank exclusion being submitted? I am not sure I fully follow the logic, as the new tests you added for such scenarios will still pass for me under the old try/except style.

I'm not saying that this isn't an improvement here, but I'm trying to understand the source of the server errors you were seeing.

Comment (higs4281) [reply wagtailautocomplete/views.py:76]: Thank your for the quick response! You raise a good question.

The new code doesn't prevent exceptions. If something evil were passed, an exception would be welcome. 

The current code prevents exceptions by silently passing on *all* of them. That's not a good practice, but it's actually irrelevant because the try/except clause just wraps a queryset definition, which hasn't been evaluated yet. So nothing has been "tried."

My goal is to make sure we don't feed a list of empty strings into our query's exclude clause, because that will cause a server error like this, which we are seeing daily:

```bash
django.db.utils.DataError: invalid input syntax for integer: ""
LINE 1: ...NOT ("ask_cfpb_answerpage"."cfgovpage_ptr_id" IN ('')) ORDER...
```

My solution is to first make sure there's an exclusion clause in the request (`if exclude`), and if so, make sure none of the exclusions are blank strings (`if item`). That way, if there are no primary keys passed, exclusions will be `[]`, which is a valid no-op exclude clause in a Django queryset. 

model.objects.exclude(pk__in=[]) is fine.
model.objects.exclude(pk__in=[""]) is a DataError. So is ["", ""]. Autocomplete is currently sending both.

Comment (higs4281) [reply wagtailautocomplete/views.py:76]: Still, why do my tests pass with the old code? I hadn't tried that.

Comment (higs4281) [reply wagtailautocomplete/views.py:76]: I'm so glad you tried my tests against the original code. 

My assumption that the queryset was not evaluated in the search view's try/except block was wrong. 
In fact the queryset gets evaluated earlier, when `getattr` is invoked on [line 69](https://github.com/wagtail/wagtail-autocomplete/blob/master/wagtailautocomplete/views.py#L69). `getattr` evaluates.

So the query is indeed evaluated when it reaches the Except clause. At that point, if empty-string exclusions are applied, a ValueError is raised and then swallowed by the blanket Exception pass. This lets the process continue, which I think is why my tests pass on the original code.

I would still advocate dropping the blanket Exception, because it can mask problems like this one.

I also realized that my tests weren't very effective because, with no pages in the test database, the queries are empty start to finish. That could mask problems, too.

So I rewrote the tests to provide pages for autocomplete to act against.

Comment (chigby) [reply wagtailautocomplete/views.py:76]: So, this makes a lot of sense to me. I'm glad my comment led you to a nicer solution here. I really like the new tests. And I agree with you that the `except Exception` is overbroad and should be removed and these errors handled more proactively. Nice work! :+1:

**Comment by higs4281** [comment]
The circleci test status response remains hung after several PR updates, and I see nothing but green success statuses at circleci.com. Is there anything I need to fix for this?
